content/blog/2021-07-14-continuous-benchmarking.md

```diff
@@ -0,0 +1,144 @@
++++
+title = "Continuous Benchmarking in Tremor"
+description = "How we implemented a continuous benchmarking pipeline in Tremor"
+date = 2021-07-14T12:18:05+05:30
+weight = 20
+draft = false
++++
+
+Recently I had the chance to implement a continuous benchmarking system for
+[Tremor](https://www.tremor.rs/) under the LFX Mentorship program.
+
+### Why?
+
+So why do we need continuous benchmarking? Let's step back and ask what's
+Tremor
+
+> Tremor is an event processing system originally designed for the needs of
+> platform engineering and infrastructure. Tremor has been running in production
+> since October 2018, processes 10 terabytes of data per day, or 10 billion
+> messages per minute and 10 million metrics per second. Tremor achieves this
+> with 10x footprint reduction in bare metal infrastructure and cloud based
+> deployments in 6 ( and counting ) systems at Wayfair, whilst reducing memory
+> usage by 10x and delivering better quality of service under conditions when
+> our network is saturated at peak eCommerce trading lifecycles. As a secondary
+> benefit, tremor is relatively low latency and relatively high throughput
+> however this is an explicit non-goal of the project. Tremor is built for users
+> that have a high message volume to deal with and want to build pipelines to
+> process, route, or limit this event stream.
+
+Tremor emerged at Wayfair because the existing data distribution tools weren’t
+performant enough for their needs. So for
+the kind of the problem Tremor is trying to solve it makes sense to track its
+performance over time and try to make it better or at least prevent any major
+regressions.[^1]
+
+[^1]: This might not be true for every change since if a there is a feature that is much needed and adding it causes a little regression. We'll probably choose adding that feature instead of rejecting it because of the regression.

```

Comment:
```suggestion
[^1]: This might not be true for every change since if there is a feature that is much needed and adding it causes a little regression. We'll probably choose adding that feature instead of rejecting it because of the regression.

```